hashing/src/lib.rs

```diff
@@ -8,6 +7,10 @@
 )]
 #![warn(missing_docs)]
 
+mod chunk_with_proof;
+mod error;
+mod indexed_merkle_proof;
```

Comment:
It looks like you want everything `#[cfg(test)]` for now, so I suggest:

```suggestion
#[cfg(test)]
mod chunk_with_proof;
#[cfg(test)]
mod error;
#[cfg(test)]
mod indexed_merkle_proof;
```

And get rid of most of the `#[cfg(test)]` annotations in `chunk_with_proof.rs` and `indexed_merkel_proof.rs`

Reply:
This will not work any longer, because with the recent changes `Digest::hash` requires `ChunkWithProof::CHUNK_SIZE`